tools/azsdk-cli/Azure.Sdk.Tools.Cli/Commands/HostServer/HostServerCommand.cs

```diff
@@ -1,26 +1,26 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 using System.CommandLine;
-using System.CommandLine.Parsing;
 using Azure.Sdk.Tools.Cli.Helpers;
 
 namespace Azure.Sdk.Tools.Cli.Commands.HostServer
 {
     public class HostServerCommand(ILogger<HostServerCommand> logger, IRawOutputHelper outputHelper)
     {
-        public static Option<string> ToolOption = new("--tools")
+        private static readonly Option<string> toolOption = new("--tools")
         {
             Description = "If provided, the mcp server will only list and respond to tools named the same as provided in this option. Glob matching is honored.",
             Required = false,
         };
 
-        public Command GetCommand()
+        public List<Command> GetCommands()
         {
-            Command cmd = new("mcp", "Starts the MCP server (stdio mode)");
-            cmd.Aliases.Add("start");  // backwards compatibility
-            cmd.Options.Add(ToolOption);
+            Command cmd = new("mcp", "Starts the MCP server (stdio mode)") { toolOption };
+            Command legacyStartCmd = new("start", "Starts the MCP server (stdio mode)") { toolOption };
+            legacyStartCmd.Hidden = true;
             cmd.SetAction((_, cancellationToken) => HandleCommand(cancellationToken));
-            return cmd;
+            legacyStartCmd.SetAction((_, cancellationToken) => HandleCommand(cancellationToken));
```

Comment:
We might need an issue for the legacy command clean-ups.

Reply:
https://github.com/Azure/azure-sdk-tools/issues/12882